### DISCUSSION TITLE
Running on Android OS

### DISCUSSION THREAD

**Discussion by mistek131995** [xjasonlyu/tun2socks#328] [General]
I put together a solution for Android OS using the commands

`
$env:CGO_ENABLED=0
$env:GOOS="android"
$env:GOARCH="arm64"
go build -o libtun2socks.so -trimpath -ldflags "-s -w -buildid=" ./
`

And tried to run the library as a process, but got an error. Maybe someone has encountered this problem and I'm doing something wrong? I'm attaching the launch code,

`
        Thread{
            val command = arrayListOf<String>(tun2socksPath(), "--proxy socks5://127.0.0.1:10808")

            val processBuilder = ProcessBuilder(command)

            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O){
                val tun2socksLog = File(applicationContext.filesDir.absolutePath + "/tun2socks.log")
                tun2socksLog.createNewFile()
                processBuilder.redirectError(tun2socksLog)
                processBuilder.redirectInput(tun2socksLog)
                processBuilder.redirectOutput(tun2socksLog)
            }

            tun2socksProcess = processBuilder.start()
            tun2socksProcess.waitFor()
        }.start()
`

I get the following error in the log:

`
flag provided but not defined: -proxy socks5://*********:10808
`

**Comment by xjasonlyu** [comment]
Somehow I think you should pass it in two strings like:  `"--proxy", "socks5://127.0.0.1:10808"`

**Reply by mistek131995** [reply]
Thanks so much) It's worked

**Comment by mistek131995** [comment]
> Somehow I think you should pass it in two strings like: `"--proxy", "socks5://127.0.0.1:10808"`

Maybe you can tell me about this problem?

time="2024-01-10T04:39:42Z" level=fatal msg="[ENGINE] failed to start: create tun: permission denied"
